stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/composer/MessageComposerViewModel.kt

```diff
@@ -97,8 +109,24 @@ public class MessageComposerViewModel(
      *
      * @param attachment The attachment to remove.
      */
-    public fun removeSelectedAttachment(attachment: Attachment): Unit =
+    public fun removeSelectedAttachment(attachment: Attachment): Unit {
         messageComposerController.removeSelectedAttachment(attachment)
+        val oldState = _messageInputState.value
+        _messageInputState.value =
+            oldState.copy(attachments = oldState.attachments - attachment,
+                validationErrors = messageComposerController.validationErrors.value)
+    }
+
+    /**
+     * Removes all the selected attachments, e.g. when user taps clear input button, or after the message is sent
```

Comment:
Is there a reason why clearing the input would also remove attachments?

Reply:
My bad, also making it private